src/subscription/PriceUtils.ts

```diff
@@ -141,4 +107,162 @@ export function getCurrentCount(featureType: BookingItemFeatureType, booking: Bo
 	} else {
 		return 0
 	}
+}
+
+
+const SUBSCRIPTION_CONFIG_RESOURCE_URL = "https://tutanota.com/resources/data/subscriptions.json"
+
+export interface PriceAndConfigProvider {
+	getSubscriptionPrice(paymentInterval: number, subscription: SubscriptionType, type: UpgradePriceType): number
+
+	getRawPricingData(): UpgradePriceServiceReturn
+
+	getSubscriptionConfig(targetSubscription: SubscriptionType): SubscriptionConfig
+
+	getSubscriptionType(lastBooking: Booking | null, customer: Customer, customerInfo: CustomerInfo): SubscriptionType
+}
+
+export async function getPricesAndConfigProvider(registrationDataId: string | null, serviceExecutor: IServiceExecutor = locator.serviceExecutor): Promise<PriceAndConfigProvider> {
+	const priceDataProvider = new HiddenPriceAndConfigProvider()
+	await priceDataProvider.init(registrationDataId, serviceExecutor)
+	return priceDataProvider
+}
+
+class HiddenPriceAndConfigProvider implements PriceAndConfigProvider {
+	private upgradePriceData: UpgradePriceServiceReturn | null = null
+	private planPrices: SubscriptionPlanPrices | null = null
+
+	private possibleSubscriptionList: { [K in SubscriptionType]: SubscriptionConfig } | null = null
+
+	async init(registrationDataId: string | null, serviceExecutor: IServiceExecutor): Promise<void> {
+		const data = createUpgradePriceServiceData({
+			date: Const.CURRENT_DATE,
+			campaign: registrationDataId,
+		})
+		this.upgradePriceData = await serviceExecutor.get(UpgradePriceService, data)
+		this.planPrices = {
+			Premium: this.upgradePriceData.premiumPrices,
+			PremiumBusiness: this.upgradePriceData.premiumBusinessPrices,
+			Teams: this.upgradePriceData.teamsPrices,
+			TeamsBusiness: this.upgradePriceData.teamsBusinessPrices,
+			Pro: this.upgradePriceData.proPrices,
+		}
+
+		if ("undefined" === typeof fetch) return
+		try {
+			this.possibleSubscriptionList = await (await fetch(SUBSCRIPTION_CONFIG_RESOURCE_URL)).json()
+		} catch (e) {
+			console.log("failed to fetch subscription list:", e)
+			throw new ConnectionError("failed to fetch subscription list")
+		}
+	}
+
+	getSubscriptionPrice(
+		paymentInterval: number,
```

Comment:
I am thinking whether we should assert more strictly and early that payment interval is only "12" and "1", what do you think?

Reply:
Technically yes, an enum would make sense.